packages/run-protocol/src/vaultFactory/liquidateIncrementally.js

```diff
@@ -0,0 +1,322 @@
+ * - Oracle quote - provide by the oracle that triggered liquidation
+ * - Oracle limit - a limit that is DIVERGENCE_TOLERANCE below the
+ *   Oracle quote.
+ * These sell based on the following:
+ * - If the AMM quote is *below* the oracle limit, skip this block
+ *   to wait for oracle and AMM prices to converge.
+ *    - should it still sell to teh Reserve here?
+ * - If the Reserve quote is higher than the AMM quote, sell to the
+ *   Reserve with a `want` of the Oracle limit
+ * - Otherwise, sell to the AMM with a `want` of the Oracle limit
+ * If selling a tranche for the minimum would pay off the debt (`isLast`),
+ * then use sell using`swapOut` with a minimum of the debt. If that fails
+ * this is not the last tranche, tehn sell with the `swapIn` operation,
+ * using the oracle price reference as the minimum.
+ *
+ * TODO integrate the reserve
+ */
+
+/** @type {ContractStartFn} */
+const start = async zcf => {
+  const { amm, priceAuthority, timerService, debtBrand } =
+    /** @type {LiquidationContractTerms} */ zcf.getTerms();
+  const nextBlock = async () => E(timerService).delay(1n);
+
+  // TODO make subject to governance
+  const SCALE = 1_000_000n;
+  const BASIS_POINTS = 10_000n * SCALE;
+  const BP2 = BASIS_POINTS * BASIS_POINTS;
+
+  const AMM_FEE_BP = 30n * SCALE;
+  const MAX_IMPACT_BP = 50n * SCALE;
+  // sqrt(impact + 1) - 1
+  // Since we take SQRT but want to be BP, multiply in BP
+  const IMPACT_FACTOR_BP =
+    BigInt(
+      Math.ceil(
+        Math.sqrt(Number((MAX_IMPACT_BP + BASIS_POINTS) * BASIS_POINTS)),
+      ),
+    ) - BASIS_POINTS;
+  const IMPACT_FACTOR_BP2 = IMPACT_FACTOR_BP * (BASIS_POINTS - AMM_FEE_BP);
+
+  const asFloat = (numerator, denominator) =>
+    Number(numerator) / Number(denominator);
+  trace(
+    'FACTORS',
+    asFloat(AMM_FEE_BP, BASIS_POINTS),
+    asFloat(MAX_IMPACT_BP, BASIS_POINTS),
+    Math.sqrt(Number(MAX_IMPACT_BP + BASIS_POINTS)),
+    asFloat(IMPACT_FACTOR_BP, BASIS_POINTS),
+    asFloat(IMPACT_FACTOR_BP2, BP2),
+  );
+
+  // How close the sale price on the AMM must be to the expected price from the oracle
+  // quote must be at least 70% 0f what the oracle says
+  const ORACLE_TOLERANCE = 3000n * SCALE;
+  const ORACLE_LIMIT_BP = makeRatio(
+    BASIS_POINTS - ORACLE_TOLERANCE,
+    debtBrand,
+    BASIS_POINTS,
+  );
+
+  // 3% max slippage = 97% min
+  const MAX_SLIPPED = makeRatio(9700n * SCALE, debtBrand, BASIS_POINTS);
+
+  trace('CONSTANTS', {
+    IMPACT_FACTOR_BP,
+    BASIS_POINTS,
+    IMPACT_FACTOR_BP2,
+    BP2,
+  });
+
+  // the maximum number fo failure sina  row before we stop liquidating.
+  // TODO make this larger andmake it reset on success
+  const MAX_FAILURES = 4;
+  let failures = 0;
+
+  const computeAMMProceeds = (poolCentral, poolCollateral, tranche, debt) => {
+    const k = NatMath.multiply(poolCentral.value, poolCollateral.value);
+    const postSaleCollateral = AmountMath.add(tranche, poolCollateral);
+    const quoteValue = NatMath.subtract(
```

Comment:
I'd call this poolCentralEstimate, I think. calling it quote made it seem like it was modelling what the price authority would say or something.

Reply:
renamed with "estimate" in most of the names